IBX-5309 - Describe Payment module (#1966)
ezsystems/developer-documentation: docs/commerce/payment/configure_payment.md

---
description: Configure the payment process.
edition: commerce
---

# Configure payment

When you work with your Commerce implementation, you can review and modify the payment configuration.

!!! note "Permissions" 

    When you modify the workflow configuration, make sure you properly set user [permissions](permission_use_cases.md#commerce) for the Payment component.

## Payment workflow

> Payment workflow relies on a [Symfony Workflow](http://symfony.com/doc/5.4/components/workflow.html).
> Each transition represents a separate payment step. 
> 
> The default fallback workflow is `ibexa_payment`.

Reviewer: ```suggestion
Payment workflow relies on a [Symfony Workflow](http://symfony.com/doc/5.4/components/workflow.html) component.
```

### Default payment workflow configuration

The default payment workflow configuration is called `ibexa_payment`, you can replace it with your custom workflow identifier if needed.

> To see the default workflow, in your project directory, navigate to the following file: `src/bundle/Resources/config/workflow.yaml`.

Author: ```suggestion
To see the default workflow, in your project directory, navigate to the following file: `vendor/Ibexa/payment/src/bundle/Resources/config/workflow.yaml`.
```

> To see the default workflow, in your project directory, navigate to the following file: `vendor/Ibexa/payment/src/bundle/Resources/config/workflow.yaml`.

Reviewer: 1) You can paste ibexa_payment workflow definition here. It will be more convenient for reader. 

2) You can add some description and even graph

Reviewer 2: In general I'd avoid directly quoting code from the product, because it causes additional maintenance overhead (https://github.com/ezsystems/developer-documentation/pull/1966#discussion_r1174840530).

> ### Custom payment workflows

Reviewer: It's important to add here configuration options which are available as part of the place/transition metadata e.g. label, translation_domain, colors etc.

> You define custom workflow implementations under the `framework.workflows` key. 
> The `payment.workflow` parameters are repository-aware.
> 
> To customize your configuration, place it in a YAML file under the `framework.workflows.<your_workflow_name>` key:

Reviewer: ```suggestion
You define custom workflow definition under the `framework.workflows` key, for example:
```

``` yaml
[[= include_file('code_samples/front/shop/payment/config/packages/ibexa.yaml', 7, 39) =]]
```

> Then reference it with `ibexa.repositories.<your_repository>.payment.workflow: your_workflow_name`, so that the system can then identify which of your configured workflows handles the payment process.

Reviewer: ```suggestion
Then reference it with `ibexa.repositories.<your_repository>.payment.workflow`, so that the system can then identify which of your configured workflows handles the payment process.
```

``` yaml
[[= include_file('code_samples/front/shop/payment/config/packages/ibexa.yaml', 0, 5) =]]
```

## Configure payment methods

You can define the payment methods [in the UI]([[= user_doc =]]/commerce/payment/configure_payment_method/).

> There is only one payment method type available: `offline`.

Reviewer: Here we need a more explanation on payment method types of put this explanation in dedicated section
